app/services/api/declarations/create.rb

```diff
@@ -0,0 +1,142 @@
+module API::Declarations
+  class Create
+    include API::Concerns::Teachers::SharedAction
+
+    attribute :declaration_date
+    attribute :declaration_type
+    attribute :evidence_type
+
+    validates :declaration_date, presence: { message: "Enter a '#/declaration_date'." }
+    validates :declaration_date,
+              future_date: true,
+              declaration_date_within_milestone: true,
+              declaration_date_format: true,
+              allow_blank: true
+    validates :declaration_type, presence: { message: "Enter a '#/declaration_type'." }
+    validates :evidence_type, evidence_type: true
+    validate :validate_only_started_or_completed_if_mentor
+    validate :teacher_not_withdrawn
+    validate :validate_milestone_exists
+    validate :validates_billable_slot_available
+    validate :payment_statement_available
+
+    def create
+      return false unless valid?
+
+      Declarations::Create.new(
+        author: Events::LeadProviderAPIAuthor.new(lead_provider:),
+        lead_provider:,
+        teacher:,
+        training_period:,
+        declaration_date:,
+        declaration_type:,
+        evidence_type:
+      ).create
+    end
+
+    def training_period
+      return unless teacher
+
+      @training_period ||= case teacher_type
+                           when :ect
+                             teacher.ect_training_periods
+                             .includes(:lead_provider)
+                             .where(active_lead_providers: { lead_provider_id: })
+                             .ongoing_or_closest_to(declaration_date)
+                             .earliest_first
```

Comment:
Not sure we need `earliest_first`? would that not just re-arrange the results again?